.github/workflows/ci-app2.yaml

```diff
@@ -19,16 +20,42 @@ jobs:
       -
         name: Set up Docker Buildx
         uses: docker/setup-buildx-action@v2
-      -
+      - 
         name: Login to DockerHub
         uses: docker/login-action@v2
         with:
           username: ${{ secrets.DOCKERHUB_USERNAME }}
           password: ${{ secrets.DOCKERHUB_TOKEN }}
+
       -
-        name: Build and push app1
+        name: Get SemVer from file
+        id: semver-reader
+        uses: juliangruber/read-file-action@v1
+        with:
+          path: ./apps/app2/.version
+          
+      - name: Parse and Load SemVer
+        run: |
+          IFS='.'
+          read -a semver <<< ${{ steps.semver-reader.outputs.content }}
+          
+          if [ -z "${semver[0]}" ] || [ -z "${semver[1]}" ] || [ -z "${semver[2]}" ]; then # basic format check
+            echo "Error: Invalid semver format."
+            exit 1
+          fi
+          
+          echo "GITHUB_SHA_VALUE=`echo $GITHUB_SHA | cut -c 9- `" >> $GITHUB_ENV # setting patch semver version value
+          echo "V_MAJOR=${semver[0]}" >> $GITHUB_ENV # setting major semver version value 
+          echo "V_MINOR=${semver[1]}" >> $GITHUB_ENV # setting minor semver version value
+          echo "V_PATCH=${semver[2]}" >> $GITHUB_ENV # setting patch semver version value
```

Comment:
make this auto increase

Reply:
Patch version number made to change based on GitHub action run number.
Also added a step to write down newest semver in .version file.
Major and minor versions to be changed manually via file.